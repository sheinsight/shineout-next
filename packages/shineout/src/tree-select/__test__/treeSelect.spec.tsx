import { render, cleanup, screen, fireEvent, waitFor } from '@testing-library/react';
import '@testing-library/jest-dom';
import TreeSelect from '..';
import mountTest from '../../tests/mountTest';
import {
  classTest,
  createClassName,
  delay,
  displayTest,
  snapshotTest,
  styleTest,
  textContentTest,
} from '../../tests/utils';
import { classLengthTest } from '../../tests/structureTest';
import TreeSelectBase from '../__example__/01-base';
import TreeMultiple from '../__example__/02-multiple';
import TreeSelectCompressed from '../__example__/03-compressed';
import TreeSelectFilter from '../__example__/04-01-filter';
import TreeSelectFilterMore from '../__example__/04-02-filter';
import TreeSelectLoader from '../__example__/05-loader';
import TreeSelectDisabled from '../__example__/06-disabled';
import TreeSelectInnerTitle from '../__example__/07-inner-title';
import TreeSelectUnmatch from '../__example__/08-unmatch';
import TreeSelectSize from '../__example__/09-size';

const SO_PREFIX = 'treeSelect';
const originClasses = ['wrapper', 'tree', 'result', 'clearIcon', 'space', 'tag', 'arrowIcon'];
const originItemClasses = ['wrapperFocus', 'arrowIconOpen'];
const { wrapper, tree, result, clearIcon, space, tag, wrapperFocus, arrowIcon, arrowIconOpen } =
  createClassName(SO_PREFIX, originClasses, originItemClasses);

const {
  icon: treeIcon,
  content: treeContent,
  root,
  node,
  children,
  text,
} = createClassName('tree', ['icon', 'content', 'children', 'root', 'node', 'text'], ['']);

const treeData = [
  {
    id: '1',
    title: 'node 1',
    children: [
      {
        id: '1-1',
        title: 'node 1-1',
        children: [
          { id: '1-1-1', title: 'node 1-1-1' },
          { id: '1-1-2', title: 'node 1-1-2' },
        ],
      },
      { id: '1-2', title: 'node 1-2' },
    ],
  },
  {
    id: '2',
    title: 'node 2',
  },
];

const TreeSelectTest = (props: any) =>{
  return (
    <TreeSelect keygen={'id'} renderItem={(d: any) => d.title} data={treeData} {...props} absolute={false} />
  )
}


beforeAll(() => {
  jest.useFakeTimers();
});
afterAll(() => {
  jest.runAllTimers();
});
afterEach(cleanup);
mountTest(<TreeSelectTest />);

describe('TreeSelect[Base]', () => {
  displayTest(TreeSelect as React.FC, 'ShineoutTreeSelect');
  snapshotTest(<TreeSelectBase />);
  snapshotTest(<TreeMultiple />, 'about multiple');
  snapshotTest(<TreeSelectCompressed />, 'about compressed');
  snapshotTest(<TreeSelectFilter />, 'about filter');
  snapshotTest(<TreeSelectFilterMore />, 'about filter more');
  snapshotTest(<TreeSelectLoader />, 'about loader');
  snapshotTest(<TreeSelectDisabled />, 'about disabled');
  snapshotTest(<TreeSelectInnerTitle />, 'about inner title');
  snapshotTest(<TreeSelectUnmatch />, 'about unmatch');
  snapshotTest(<TreeSelectSize />, 'about size');
  test('should render when set className and style', () => {
    const className = 'test';
    const style = { color: 'red' };
    const styleStr = 'color: red;';
    const { container } = render(<TreeSelectTest className={className} style={style} />);
    const selectWrapper = container.querySelector(wrapper)!;
    classTest(selectWrapper, className);
    styleTest(selectWrapper, styleStr);
  });
  // refer to select test
  test('should render default', async () => {
    const { container } = render(<TreeSelectTest />);
    const selectWrapper = container.querySelector(wrapper)!;
    fireEvent.click(selectWrapper.querySelector(result)!)
    await waitFor(async () => {
      await delay(200);
    });

    classLengthTest(container.querySelector(wrapper)!, tree, 1);
  });
  test('should render when advanced filtering', async () => {
    const filterText = '1-1';
    const { container } = render(
      <TreeSelectTest
        showHitDescendants
        onAdvancedFilter={(text: any) => (d: any) => d.title.indexOf(text) > -1}
        clearable
      />,
    );
    const selectWrapper = container.querySelector(wrapper)!;
    const treeResult = selectWrapper.querySelector(result)!;
    fireEvent.click(treeResult);
    await waitFor(async () => {
      await delay(200);
    });
    fireEvent.change(selectWrapper.querySelector('input')!, { target: { value: filterText } });
    await waitFor(async () => {
      await delay(500);
    });
    const treeIconWrapper = container.querySelector(treeIcon)!;
    // styleTest(treeIconWrapper.firstElementChild!, filterTreeIconStyle);
    expect(screen.getByText(filterText)).toBeInTheDocument();
    classLengthTest(container, treeContent, 4);
    fireEvent.click(treeIconWrapper.querySelector('svg')!);
    await waitFor(async () => {
      await delay(200);
    });
    classLengthTest(container, treeContent, 5);
  });
  test('should render when set expand', async () => {
    const { container } = render(<TreeSelectTest expanded={['1']} />);
    const selectWrapper = container.querySelector(wrapper)!;
    fireEvent.click(selectWrapper.querySelector(result)!)
    await waitFor(async () => {
      await delay(200);
    });
    const treeRoot = container.querySelector(root)!;
    const treeNodes = treeRoot.querySelectorAll(`:scope > ${node}`);
    expect(treeNodes.length).toBe(2);
    classLengthTest(treeRoot, children, 1);
  });
  test('should render when set getComponentRef', () => {
    const getComponentRef = jest.fn();
    render(<TreeSelectTest getComponentRef={getComponentRef} />);
    expect(getComponentRef.mock.calls.length).toBe(1);
  });
  test('should render when click single', async () => {
    const onChangeAddition = jest.fn();
    const { container } = render(<TreeSelectTest onChangeAddition={onChangeAddition} />);
    const selectWrapper = container.querySelector(wrapper)!;
    fireEvent.click(selectWrapper.querySelector(result)!)
    await waitFor(async () => {
      await delay(200);
    });
    const treeRoot = selectWrapper.querySelector(root)!;
    const treeNodes = treeRoot.querySelectorAll(`:scope > ${node}`);
    const treeResult = selectWrapper.querySelector(result)!;
    fireEvent.click(treeResult);
    await waitFor(async () => {
      await delay(200);
    });
    fireEvent.click(treeNodes[0].querySelector('svg')!);
    const treeChildren = treeNodes[0].querySelector(children)!;
    const treeChildrenNodes = treeChildren.querySelectorAll(node);
    fireEvent.click(treeChildrenNodes[1].querySelector(text)!);
    await waitFor(async () => {
      await delay(200);
    });
    textContentTest(treeResult, treeData?.[0]?.children?.[1]?.title);
    classLengthTest(treeResult, space, 0);
    fireEvent.mouseEnter(treeResult);
    await waitFor(async () => {
      await delay(200);
    });
    fireEvent.click(selectWrapper.querySelector(clearIcon)!);
    await waitFor(async () => {
      await delay(200);
    });
    classLengthTest(treeResult, space, 1);
    expect(onChangeAddition.mock.calls.length).toBe(2);
    fireEvent.mouseLeave(treeResult);
  });
  test('should render when click multiple', async () => {
    const { container } = render(<TreeSelectTest multiple />);
    const selectWrapper = container.querySelector(wrapper)!;
    fireEvent.click(selectWrapper.querySelector(result)!)
    await waitFor(async () => {
      await delay(200);
    });
    const treeRoot = selectWrapper.querySelector(root)!;
    const treeNodes = treeRoot.querySelectorAll(`:scope > ${node}`);
    const treeResult = selectWrapper.querySelector(result)!;
    fireEvent.click(treeResult);
    await waitFor(async () => {
      await delay(200);
    });
    fireEvent.click(treeNodes[0].querySelector('svg')!);
    const treeChildren = treeNodes[0].querySelector(children)!;
    const treeChildrenNodes = treeChildren.querySelectorAll(node);
    fireEvent.click(treeChildrenNodes[1].querySelector('input')!);
    fireEvent.click(treeNodes[1].querySelector('input')!);
    await waitFor(async () => {
      await delay(200);
    });
    const selectTags = treeResult.querySelectorAll(tag);
    expect(selectTags.length).toBe(3);
    fireEvent.click(selectTags[0].querySelector('svg')!);
    await waitFor(async () => {
      await delay(200);
    });
    expect(treeResult.querySelectorAll(tag).length).toBe(1);
  });
  test('should render when focus and blur', async () => {
    const { container } = render(<TreeSelectTest />);
    const selectWrapper = container.querySelector(wrapper)!;
    fireEvent.focus(selectWrapper);
    await waitFor(async () => {
      await delay(200);
    });
    classTest(selectWrapper, wrapperFocus);
    fireEvent.blur(selectWrapper);
    await waitFor(async () => {
      await delay(200);
    });
    // console.log('selectWrapper classList: >>', selectWrapper?.classList?.toString())
    classTest(selectWrapper, wrapperFocus, false);
  });
  test('should render when keyDown and tab', async () => {
    const onEnterExpand = jest.fn();
    const { container } = render(<TreeSelectTest onEnterExpand={onEnterExpand} open={undefined} />);
    const selectWrapper = container.querySelector(wrapper)!;
    fireEvent.keyDown(selectWrapper, { keyCode: 13 });
    fireEvent.click(selectWrapper.querySelector(result)!)
    await waitFor(async () => {
      await delay(200);
    });
    classTest(selectWrapper.querySelector(arrowIcon)!, arrowIconOpen);
    fireEvent.keyDown(selectWrapper, { keyCode: 9 });
    await waitFor(async () => {
      await delay(200);
    });
    const iconElement =   selectWrapper.querySelector(arrowIcon)!;
    // console.log('iconElement classList >>', iconElement.classList.toString())
    classTest(iconElement, arrowIconOpen, false);
    expect(onEnterExpand.mock.calls.length).toBe(1);
  });
});
