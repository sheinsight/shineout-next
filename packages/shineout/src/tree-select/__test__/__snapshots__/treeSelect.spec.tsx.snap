// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TreeSelect[Base] should render correctly  1`] = `
<div>
  <div
    class="c-rcwgjo soui-treeSelect-wrapper c-un6lio soui-treeSelect-clearable"
    data-soui-input-border="true"
    style="width: 300px;"
    tabindex="0"
  >
    <div
      class="c-ccl0f5 soui-treeSelect-result-wrapper c-m6ehm2 soui-treeSelect-wrapper-padding-box cwmkarb soui-treeSelect-wrapper-inner-title-top c-b842ev soui-treeSelect-wrapper-inner-title-bottom"
    >
      <div
        class="ch6kcpg soui-treeSelect-result"
      >
        <div
          class="cst20xc soui-select-result-text-wrapper"
        >
          <span
            class="c-mt24py soui-select-placeholder c-pjllh2 soui-select-ellipsis"
          >
            <span>
              Please select content
            </span>
          </span>
        </div>
      </div>
    </div>
    <span
      class="c-qlrywl soui-treeSelect-arrow-icon"
    >
      <svg
        aria-hidden="true"
        fill="currentColor"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M11.2929 15.7071C11.6534 16.0676 12.2206 16.0953 12.6129 15.7903L12.7071 15.7071L18.7071 9.70711C19.0976 9.31658 19.0976 8.68342 18.7071 8.29289C18.3466 7.93241 17.7794 7.90468 17.3871 8.2097L17.2929 8.29289L12.7072 12.8779C12.3165 13.2683 11.6835 13.2683 11.2928 12.8779L6.70711 8.29289C6.34662 7.93241 5.77939 7.90468 5.3871 8.2097L5.29289 8.29289C4.93241 8.65338 4.90468 9.22061 5.2097 9.6129L5.29289 9.70711L11.2929 15.7071Z"
        />
      </svg>
    </span>
  </div>
</div>
`;

exports[`TreeSelect[Base] should render correctly about compressed 1`] = `
<div>
  <div
    class="c-rcwgjo soui-treeSelect-wrapper c-un6lio soui-treeSelect-clearable c-lkqjcp soui-treeSelect-multiple"
    data-soui-input-border="true"
    style="width: 300px;"
    tabindex="0"
  >
    <div
      class="c-ccl0f5 soui-treeSelect-result-wrapper c-m6ehm2 soui-treeSelect-wrapper-padding-box cwmkarb soui-treeSelect-wrapper-inner-title-top c-b842ev soui-treeSelect-wrapper-inner-title-bottom"
    >
      <div
        class="ch6kcpg soui-treeSelect-result"
      >
        <div
          class="cst20xc soui-select-result-text-wrapper cd43w9 soui-select-compressed-wrapper c-7vjtar soui-select-multiple-result-wrapper c-7a22yf soui-select-multiple-compressed-wrapper"
        >
          <span
            class="c-mt24py soui-select-placeholder c-pjllh2 soui-select-ellipsis"
          >
            <span>
              Please select content
            </span>
          </span>
        </div>
      </div>
    </div>
    <span
      class="c-qlrywl soui-treeSelect-arrow-icon"
    >
      <svg
        aria-hidden="true"
        fill="currentColor"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M5 10C6.10457 10 7 10.8954 7 12C7 13.1046 6.10457 14 5 14C3.89543 14 3 13.1046 3 12C3 10.8954 3.89543 10 5 10ZM12 10C13.1046 10 14 10.8954 14 12C14 13.1046 13.1046 14 12 14C10.8954 14 10 13.1046 10 12C10 10.8954 10.8954 10 12 10ZM19 10C20.1046 10 21 10.8954 21 12C21 13.1046 20.1046 14 19 14C17.8954 14 17 13.1046 17 12C17 10.8954 17.8954 10 19 10Z"
        />
      </svg>
    </span>
  </div>
</div>
`;

exports[`TreeSelect[Base] should render correctly about disabled 1`] = `
<div>
  <div
    class="c-rcwgjo soui-treeSelect-wrapper c-8p7oko soui-treeSelect-wrapper-disabled c-un6lio soui-treeSelect-clearable"
    data-soui-input-border="true"
    style="width: 250px;"
    tabindex="-1"
  >
    <div
      class="c-ccl0f5 soui-treeSelect-result-wrapper c-m6ehm2 soui-treeSelect-wrapper-padding-box cwmkarb soui-treeSelect-wrapper-inner-title-top c-b842ev soui-treeSelect-wrapper-inner-title-bottom"
    >
      <div
        class="ch6kcpg soui-treeSelect-result"
      >
        <div
          class="cst20xc soui-select-result-text-wrapper"
        >
          <span
            class="c-mt24py soui-select-placeholder c-pjllh2 soui-select-ellipsis"
          >
            <span>
              Please select content
            </span>
          </span>
        </div>
      </div>
    </div>
    <span
      class="c-qlrywl soui-treeSelect-arrow-icon"
    >
      <svg
        aria-hidden="true"
        fill="currentColor"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M11.2929 15.7071C11.6534 16.0676 12.2206 16.0953 12.6129 15.7903L12.7071 15.7071L18.7071 9.70711C19.0976 9.31658 19.0976 8.68342 18.7071 8.29289C18.3466 7.93241 17.7794 7.90468 17.3871 8.2097L17.2929 8.29289L12.7072 12.8779C12.3165 13.2683 11.6835 13.2683 11.2928 12.8779L6.70711 8.29289C6.34662 7.93241 5.77939 7.90468 5.3871 8.2097L5.29289 8.29289C4.93241 8.65338 4.90468 9.22061 5.2097 9.6129L5.29289 9.70711L11.2929 15.7071Z"
        />
      </svg>
    </span>
  </div>
  <div
    class="c-rcwgjo soui-treeSelect-wrapper c-un6lio soui-treeSelect-clearable"
    data-soui-input-border="true"
    style="width: 300px; margin-left: 24px;"
    tabindex="0"
  >
    <div
      class="c-ccl0f5 soui-treeSelect-result-wrapper c-m6ehm2 soui-treeSelect-wrapper-padding-box cwmkarb soui-treeSelect-wrapper-inner-title-top c-b842ev soui-treeSelect-wrapper-inner-title-bottom"
    >
      <div
        class="ch6kcpg soui-treeSelect-result"
      >
        <div
          class="cst20xc soui-select-result-text-wrapper"
        >
          <span
            class="c-mt24py soui-select-placeholder c-pjllh2 soui-select-ellipsis"
          >
            <span>
              Please select content
            </span>
          </span>
        </div>
      </div>
    </div>
    <span
      class="c-qlrywl soui-treeSelect-arrow-icon"
    >
      <svg
        aria-hidden="true"
        fill="currentColor"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M11.2929 15.7071C11.6534 16.0676 12.2206 16.0953 12.6129 15.7903L12.7071 15.7071L18.7071 9.70711C19.0976 9.31658 19.0976 8.68342 18.7071 8.29289C18.3466 7.93241 17.7794 7.90468 17.3871 8.2097L17.2929 8.29289L12.7072 12.8779C12.3165 13.2683 11.6835 13.2683 11.2928 12.8779L6.70711 8.29289C6.34662 7.93241 5.77939 7.90468 5.3871 8.2097L5.29289 8.29289C4.93241 8.65338 4.90468 9.22061 5.2097 9.6129L5.29289 9.70711L11.2929 15.7071Z"
        />
      </svg>
    </span>
  </div>
</div>
`;

exports[`TreeSelect[Base] should render correctly about filter 1`] = `
<div>
  <div
    class="c-rcwgjo soui-treeSelect-wrapper c-un6lio soui-treeSelect-clearable"
    data-soui-input-border="true"
    style="width: 300px;"
    tabindex="0"
  >
    <div
      class="c-ccl0f5 soui-treeSelect-result-wrapper c-m6ehm2 soui-treeSelect-wrapper-padding-box cwmkarb soui-treeSelect-wrapper-inner-title-top c-b842ev soui-treeSelect-wrapper-inner-title-bottom"
    >
      <div
        class="ch6kcpg soui-treeSelect-result"
      >
        <div
          class="cst20xc soui-select-result-text-wrapper"
        >
          <span
            class="c-mt24py soui-select-placeholder c-pjllh2 soui-select-ellipsis"
          >
            <span>
              Please select content
            </span>
          </span>
        </div>
      </div>
    </div>
    <span
      class="c-qlrywl soui-treeSelect-arrow-icon"
    >
      <svg
        aria-hidden="true"
        fill="currentColor"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M11.2929 15.7071C11.6534 16.0676 12.2206 16.0953 12.6129 15.7903L12.7071 15.7071L18.7071 9.70711C19.0976 9.31658 19.0976 8.68342 18.7071 8.29289C18.3466 7.93241 17.7794 7.90468 17.3871 8.2097L17.2929 8.29289L12.7072 12.8779C12.3165 13.2683 11.6835 13.2683 11.2928 12.8779L6.70711 8.29289C6.34662 7.93241 5.77939 7.90468 5.3871 8.2097L5.29289 8.29289C4.93241 8.65338 4.90468 9.22061 5.2097 9.6129L5.29289 9.70711L11.2929 15.7071Z"
        />
      </svg>
    </span>
  </div>
</div>
`;

exports[`TreeSelect[Base] should render correctly about filter more 1`] = `
<div>
  <div
    class="c-rcwgjo soui-treeSelect-wrapper c-un6lio soui-treeSelect-clearable"
    data-soui-input-border="true"
    style="width: 300px;"
    tabindex="0"
  >
    <div
      class="c-ccl0f5 soui-treeSelect-result-wrapper c-m6ehm2 soui-treeSelect-wrapper-padding-box cwmkarb soui-treeSelect-wrapper-inner-title-top c-b842ev soui-treeSelect-wrapper-inner-title-bottom"
    >
      <div
        class="ch6kcpg soui-treeSelect-result"
      >
        <div
          class="cst20xc soui-select-result-text-wrapper"
        >
          <span
            class="c-mt24py soui-select-placeholder c-pjllh2 soui-select-ellipsis"
          >
            <span>
              Please select content
            </span>
          </span>
        </div>
      </div>
    </div>
    <span
      class="c-qlrywl soui-treeSelect-arrow-icon"
    >
      <svg
        aria-hidden="true"
        fill="currentColor"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M11.2929 15.7071C11.6534 16.0676 12.2206 16.0953 12.6129 15.7903L12.7071 15.7071L18.7071 9.70711C19.0976 9.31658 19.0976 8.68342 18.7071 8.29289C18.3466 7.93241 17.7794 7.90468 17.3871 8.2097L17.2929 8.29289L12.7072 12.8779C12.3165 13.2683 11.6835 13.2683 11.2928 12.8779L6.70711 8.29289C6.34662 7.93241 5.77939 7.90468 5.3871 8.2097L5.29289 8.29289C4.93241 8.65338 4.90468 9.22061 5.2097 9.6129L5.29289 9.70711L11.2929 15.7071Z"
        />
      </svg>
    </span>
  </div>
</div>
`;

exports[`TreeSelect[Base] should render correctly about inner title 1`] = `
<div>
  <div
    class="c-rcwgjo soui-treeSelect-wrapper c-a7xog2 soui-treeSelect-wrapper-inner-title c-un6lio soui-treeSelect-clearable"
    data-soui-input-border="true"
    style="width: 300px;"
    tabindex="0"
  >
    <div
      class="c-ccl0f5 soui-treeSelect-result-wrapper c-m6ehm2 soui-treeSelect-wrapper-padding-box cwmkarb soui-treeSelect-wrapper-inner-title-top c-b842ev soui-treeSelect-wrapper-inner-title-bottom"
    >
      <div
        class="c-o6szqp soui-inner-title-wrapper"
      >
        <div
          class="c4kxj2s soui-inner-title-title cv1qspd soui-inner-title-top"
        >
          Please select content
        </div>
        <div
          class="c4irvcz soui-inner-title-place"
        >
          <div
            class="c4kxj2s soui-inner-title-title"
          >
            Please select content
          </div>
        </div>
        <div
          class="c-yu7nrf soui-inner-title-content"
        >
          <div
            class="ch6kcpg soui-treeSelect-result"
          >
            <div
              class="cst20xc soui-select-result-text-wrapper"
            >
              <div
                class="ccborcd soui-select-space"
              >
                 
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <span
      class="c-qlrywl soui-treeSelect-arrow-icon"
    >
      <svg
        aria-hidden="true"
        fill="currentColor"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M11.2929 15.7071C11.6534 16.0676 12.2206 16.0953 12.6129 15.7903L12.7071 15.7071L18.7071 9.70711C19.0976 9.31658 19.0976 8.68342 18.7071 8.29289C18.3466 7.93241 17.7794 7.90468 17.3871 8.2097L17.2929 8.29289L12.7072 12.8779C12.3165 13.2683 11.6835 13.2683 11.2928 12.8779L6.70711 8.29289C6.34662 7.93241 5.77939 7.90468 5.3871 8.2097L5.29289 8.29289C4.93241 8.65338 4.90468 9.22061 5.2097 9.6129L5.29289 9.70711L11.2929 15.7071Z"
        />
      </svg>
    </span>
  </div>
</div>
`;

exports[`TreeSelect[Base] should render correctly about loader 1`] = `
<div
  class="c-rcwgjo soui-treeSelect-wrapper c-un6lio soui-treeSelect-clearable c-lkqjcp soui-treeSelect-multiple"
  data-soui-input-border="true"
  style="width: 300px;"
  tabindex="0"
>
  <div
    class="c-ccl0f5 soui-treeSelect-result-wrapper c-m6ehm2 soui-treeSelect-wrapper-padding-box cwmkarb soui-treeSelect-wrapper-inner-title-top c-b842ev soui-treeSelect-wrapper-inner-title-bottom"
  >
    <div
      class="ch6kcpg soui-treeSelect-result"
    >
      <div
        class="cst20xc soui-select-result-text-wrapper c-7vjtar soui-select-multiple-result-wrapper"
      >
        <span
          class="c-mt24py soui-select-placeholder c-pjllh2 soui-select-ellipsis"
        >
          <span>
            Please select content
          </span>
        </span>
      </div>
    </div>
  </div>
  <span
    class="c-qlrywl soui-treeSelect-arrow-icon"
  >
    <svg
      aria-hidden="true"
      fill="currentColor"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M11.2929 15.7071C11.6534 16.0676 12.2206 16.0953 12.6129 15.7903L12.7071 15.7071L18.7071 9.70711C19.0976 9.31658 19.0976 8.68342 18.7071 8.29289C18.3466 7.93241 17.7794 7.90468 17.3871 8.2097L17.2929 8.29289L12.7072 12.8779C12.3165 13.2683 11.6835 13.2683 11.2928 12.8779L6.70711 8.29289C6.34662 7.93241 5.77939 7.90468 5.3871 8.2097L5.29289 8.29289C4.93241 8.65338 4.90468 9.22061 5.2097 9.6129L5.29289 9.70711L11.2929 15.7071Z"
      />
    </svg>
  </span>
</div>
`;

exports[`TreeSelect[Base] should render correctly about multiple 1`] = `
<div>
  <div
    class="c-rcwgjo soui-treeSelect-wrapper c-un6lio soui-treeSelect-clearable c-lkqjcp soui-treeSelect-multiple"
    data-soui-input-border="true"
    style="width: 300px;"
    tabindex="0"
  >
    <div
      class="c-ccl0f5 soui-treeSelect-result-wrapper c-m6ehm2 soui-treeSelect-wrapper-padding-box cwmkarb soui-treeSelect-wrapper-inner-title-top c-b842ev soui-treeSelect-wrapper-inner-title-bottom"
    >
      <div
        class="ch6kcpg soui-treeSelect-result"
      >
        <div
          class="cst20xc soui-select-result-text-wrapper c-7vjtar soui-select-multiple-result-wrapper"
        >
          <span
            class="c-mt24py soui-select-placeholder c-pjllh2 soui-select-ellipsis"
          >
            <span>
              Please select content
            </span>
          </span>
        </div>
      </div>
    </div>
    <span
      class="c-qlrywl soui-treeSelect-arrow-icon"
    >
      <svg
        aria-hidden="true"
        fill="currentColor"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M11.2929 15.7071C11.6534 16.0676 12.2206 16.0953 12.6129 15.7903L12.7071 15.7071L18.7071 9.70711C19.0976 9.31658 19.0976 8.68342 18.7071 8.29289C18.3466 7.93241 17.7794 7.90468 17.3871 8.2097L17.2929 8.29289L12.7072 12.8779C12.3165 13.2683 11.6835 13.2683 11.2928 12.8779L6.70711 8.29289C6.34662 7.93241 5.77939 7.90468 5.3871 8.2097L5.29289 8.29289C4.93241 8.65338 4.90468 9.22061 5.2097 9.6129L5.29289 9.70711L11.2929 15.7071Z"
        />
      </svg>
    </span>
  </div>
</div>
`;

exports[`TreeSelect[Base] should render correctly about size 1`] = `
<div
  style="width: 100%; display: flex; align-items: center; gap: 24px;"
>
  <div
    class="c-rcwgjo soui-treeSelect-wrapper c-k2l5n9 soui-treeSelect-wrapper-small c-un6lio soui-treeSelect-clearable"
    data-soui-input-border="true"
    style="width: 300px;"
    tabindex="0"
  >
    <div
      class="c-ccl0f5 soui-treeSelect-result-wrapper c-m6ehm2 soui-treeSelect-wrapper-padding-box cwmkarb soui-treeSelect-wrapper-inner-title-top c-b842ev soui-treeSelect-wrapper-inner-title-bottom"
    >
      <div
        class="ch6kcpg soui-treeSelect-result"
      >
        <div
          class="cst20xc soui-select-result-text-wrapper"
        >
          <span
            class="c-mt24py soui-select-placeholder c-pjllh2 soui-select-ellipsis"
          >
            <span>
              Please select content
            </span>
          </span>
        </div>
      </div>
    </div>
    <span
      class="c-qlrywl soui-treeSelect-arrow-icon"
    >
      <svg
        aria-hidden="true"
        fill="currentColor"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M11.2929 15.7071C11.6534 16.0676 12.2206 16.0953 12.6129 15.7903L12.7071 15.7071L18.7071 9.70711C19.0976 9.31658 19.0976 8.68342 18.7071 8.29289C18.3466 7.93241 17.7794 7.90468 17.3871 8.2097L17.2929 8.29289L12.7072 12.8779C12.3165 13.2683 11.6835 13.2683 11.2928 12.8779L6.70711 8.29289C6.34662 7.93241 5.77939 7.90468 5.3871 8.2097L5.29289 8.29289C4.93241 8.65338 4.90468 9.22061 5.2097 9.6129L5.29289 9.70711L11.2929 15.7071Z"
        />
      </svg>
    </span>
  </div>
  <div
    class="c-rcwgjo soui-treeSelect-wrapper c-un6lio soui-treeSelect-clearable"
    data-soui-input-border="true"
    style="width: 300px;"
    tabindex="0"
  >
    <div
      class="c-ccl0f5 soui-treeSelect-result-wrapper c-m6ehm2 soui-treeSelect-wrapper-padding-box cwmkarb soui-treeSelect-wrapper-inner-title-top c-b842ev soui-treeSelect-wrapper-inner-title-bottom"
    >
      <div
        class="ch6kcpg soui-treeSelect-result"
      >
        <div
          class="cst20xc soui-select-result-text-wrapper"
        >
          <span
            class="c-mt24py soui-select-placeholder c-pjllh2 soui-select-ellipsis"
          >
            <span>
              Please select content
            </span>
          </span>
        </div>
      </div>
    </div>
    <span
      class="c-qlrywl soui-treeSelect-arrow-icon"
    >
      <svg
        aria-hidden="true"
        fill="currentColor"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M11.2929 15.7071C11.6534 16.0676 12.2206 16.0953 12.6129 15.7903L12.7071 15.7071L18.7071 9.70711C19.0976 9.31658 19.0976 8.68342 18.7071 8.29289C18.3466 7.93241 17.7794 7.90468 17.3871 8.2097L17.2929 8.29289L12.7072 12.8779C12.3165 13.2683 11.6835 13.2683 11.2928 12.8779L6.70711 8.29289C6.34662 7.93241 5.77939 7.90468 5.3871 8.2097L5.29289 8.29289C4.93241 8.65338 4.90468 9.22061 5.2097 9.6129L5.29289 9.70711L11.2929 15.7071Z"
        />
      </svg>
    </span>
  </div>
  <div
    class="c-rcwgjo soui-treeSelect-wrapper c-k6n15t soui-treeSelect-wrapper-large c-un6lio soui-treeSelect-clearable"
    data-soui-input-border="true"
    style="width: 300px;"
    tabindex="0"
  >
    <div
      class="c-ccl0f5 soui-treeSelect-result-wrapper c-m6ehm2 soui-treeSelect-wrapper-padding-box cwmkarb soui-treeSelect-wrapper-inner-title-top c-b842ev soui-treeSelect-wrapper-inner-title-bottom"
    >
      <div
        class="ch6kcpg soui-treeSelect-result"
      >
        <div
          class="cst20xc soui-select-result-text-wrapper"
        >
          <span
            class="c-mt24py soui-select-placeholder c-pjllh2 soui-select-ellipsis"
          >
            <span>
              Please select content
            </span>
          </span>
        </div>
      </div>
    </div>
    <span
      class="c-qlrywl soui-treeSelect-arrow-icon"
    >
      <svg
        aria-hidden="true"
        fill="currentColor"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M11.2929 15.7071C11.6534 16.0676 12.2206 16.0953 12.6129 15.7903L12.7071 15.7071L18.7071 9.70711C19.0976 9.31658 19.0976 8.68342 18.7071 8.29289C18.3466 7.93241 17.7794 7.90468 17.3871 8.2097L17.2929 8.29289L12.7072 12.8779C12.3165 13.2683 11.6835 13.2683 11.2928 12.8779L6.70711 8.29289C6.34662 7.93241 5.77939 7.90468 5.3871 8.2097L5.29289 8.29289C4.93241 8.65338 4.90468 9.22061 5.2097 9.6129L5.29289 9.70711L11.2929 15.7071Z"
        />
      </svg>
    </span>
  </div>
</div>
`;

exports[`TreeSelect[Base] should render correctly about unmatch 1`] = `
<div>
  <div
    class="c-rcwgjo soui-treeSelect-wrapper c-un6lio soui-treeSelect-clearable"
    data-soui-input-border="true"
    style="width: 300px;"
    tabindex="0"
  >
    <div
      class="c-ccl0f5 soui-treeSelect-result-wrapper c-m6ehm2 soui-treeSelect-wrapper-padding-box cwmkarb soui-treeSelect-wrapper-inner-title-top c-b842ev soui-treeSelect-wrapper-inner-title-bottom"
    >
      <div
        class="ch6kcpg soui-treeSelect-result"
      >
        <div
          class="cst20xc soui-select-result-text-wrapper"
        >
          <span
            class="c-mt24py soui-select-placeholder c-pjllh2 soui-select-ellipsis"
          >
            <span>
              Please select content
            </span>
          </span>
        </div>
      </div>
    </div>
    <span
      class="c-qlrywl soui-treeSelect-arrow-icon"
    >
      <svg
        aria-hidden="true"
        fill="currentColor"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M11.2929 15.7071C11.6534 16.0676 12.2206 16.0953 12.6129 15.7903L12.7071 15.7071L18.7071 9.70711C19.0976 9.31658 19.0976 8.68342 18.7071 8.29289C18.3466 7.93241 17.7794 7.90468 17.3871 8.2097L17.2929 8.29289L12.7072 12.8779C12.3165 13.2683 11.6835 13.2683 11.2928 12.8779L6.70711 8.29289C6.34662 7.93241 5.77939 7.90468 5.3871 8.2097L5.29289 8.29289C4.93241 8.65338 4.90468 9.22061 5.2097 9.6129L5.29289 9.70711L11.2929 15.7071Z"
        />
      </svg>
    </span>
  </div>
</div>
`;
